### PR TITLE
Nag at uses of Provider.forUseAtConfigurationTime

### LINKED ISSUE #21796: Report deprecation of Provider.forUseAtConfigurationTime()
It is already soft deprecated (`@Deprecated` annotation).
This issue is about adding deprecation reporting/logging for invocations.

This issue needs to be fixed and a KGP version available with the fix before we can add the nagging.

* [KT-53823 - Kotlin Gradle Plugin uses deprecated Gradle API: Provider.forUseAtConfigurationTime()
](https://youtrack.jetbrains.com/issue/KT-53823)

This issue in the Protobuf plugin needs to be fixed first:
* https://github.com/google/osdetector-gradle-plugin/issues/29

All requirements have been met, this is actionable!

### PR DESCRIPTION
**Pull request by mlopatkin**
This has been soft-deprecated since Gradle 7.4 and is a no-op.

Fixes #21796.

### CODE REVIEW

**Comment by mlopatkin** [comment]
@bot-gradle test this

**Comment by mlopatkin** [comment]
@bot-gradle test this

**Comment by mlopatkin** [comment]
@bot-gradle test this

File: subprojects/language-java/src/integTest/groovy/org/gradle/api/tasks/JavaToolchainBuildOperationsIntegrationTest.groovy [hunk 1]
```diff
@@ -469,6 +469,16 @@ class JavaToolchainBuildOperationsIntegrationTest extends AbstractIntegrationSpe
                     "Consult the upgrading guide for further information: " +
                     "https://docs.gradle.org/current/userguide/upgrading_version_8.html#deprecated_access_to_conventions")
         }
+        if (!isKotlin1dot8) {
+            if (GradleContextualExecuter.isConfigCache()) {
+                executer.expectDocumentedDeprecationWarning(
```
Comment (bamboo) [inline subprojects/language-java/src/integTest/groovy/org/gradle/api/tasks/JavaToolchainBuildOperationsIntegrationTest.groovy:474]: ❓ Not possible to reuse `expectForUseAtConfigurationTimeDeprecation` here?

**Review by bamboo — APPROVED** [review]

**Review by bamboo — APPROVED** [review]

**Comment by mlopatkin** [comment]
@bot-gradle test and merge

**Pull request merged**